test/bzl_test.bzl

```diff
@@ -0,0 +1,42 @@
+"""Macro for Ensuring Starlark Dependencies are Specified Properly"""
+
+load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+load("@io_bazel_stardoc//stardoc:stardoc.bzl", "stardoc")
+
+def bzl_test(name, src, deps):
+    """Provides build-time assurances that `bzl_library` declarations exist and are referenced properly.
+
+    This macro relies upon Stardoc's ability to traverse `bzl_library`
+    dependencies. If a Starlark dependency is loaded, but not specified as a
+    dependency, the Stardoc utility will fail with a reasonably helpful error
+    message. Interestingly, the Stardoc utility does not apply the same rigor
+    to files that are directly specifed to it.
```

Comment:
Heh, yeah. I'm pretty sure the starlark environment that stardoc creates is subtly different than real bazel. I've spent a few hours today trying to do some feature detection because stardoc seems to not have native.bazel_version in certain contexts :facepalm:

Reply:
Are you running stardoc with Bazel 7? It should no longer use a fake environment with that version.

Reply:
@fmeum If you are asking me about stardoc and Bazel 7, I wrote `bzl_test` for 6.x.